tests/profiler/test_profiler.py

```diff
@@ -593,7 +593,10 @@ def test_pytorch_profiler_raises_warning_for_limited_steps(tmpdir, trainer_confi
     warning_cache.clear()
     with pytest.warns(UserWarning, match="not enough steps to properly record traces"):
         getattr(trainer, trainer_fn)(model)
-        assert trainer.profiler._schedule is None
+        if hasattr(trainer.profiler, "_schedule"):
+            assert trainer.profiler._schedule is None
+        elif hasattr(trainer.profiler, "_override_steps"):
```

Comment:
Why do we need these `if hasattr` checks? I don't like them because if the attributes are removed from the classes, then these lines will stay here unnoticed forever.